src/components/UI/SearchInput.js

```diff
@@ -0,0 +1,53 @@
+import styles from "./SearchInput.module.css";
+import { useEffect, useState } from "react";
+import { useRef } from "react";
+
+const DEFAULT_DELAY = 1000;
+
+const SearchInput = ({ onChange, delay }) => {
+  const [value, setValue] = useState("");
+  const [isEmpty, setIsEmpty] = useState(true);
```

Comment:
You don't need this state unless you need track "touched" inputs

`value !== ''` is a same state.